src/data/process.py

```diff
@@ -79,81 +91,52 @@ def parse_articles_data(source, data_raw_dir, verbose=False):
-    """Parse pubmed xml article data and return metadata and text."""
-    metadata = pubmed_parser.parse_pubmed_xml(src)
-    text = pubmed_parser.parse_pubmed_paragraph(src, all_paragraph=True)
-    text = ' '.join(' '.join([x['text'] for x in text]).split())
-    return metadata, text
-
+def parse_html(src):
+    """Parse research paper HTML and return text."""
+    with open(src, 'r', encoding='utf-8') as f:
+        raw_data_ = f.read()
+    soup = bs(raw_data_)
+    try:
+        text = ' '.join(soup.find('article').get_text().split())
+    except AttributeError:
+        text = None
+    return text
 
-def aggregate_pubmed(srcs, verbose=False):
-    """Aggregate a set of Pubmed article text and metadata."""
-    pubmed_data = OrderedDict()
-    pubmed_metadata = OrderedDict()
-    for src in srcs:
-        if verbose:
-            print(src)
-        try:
-            temp = OrderedDict()
-            metadata, text = parse_pubmed(str(src.absolute()))
-            temp['metadata'] = metadata
-            temp['metadata']['source'] = 'pubmed'
-            temp['text'] = text
-            try:
-                k = metadata['doi']
-            except AttributeError:
-                k = src.strip('.xml')
-            if len(text) > 0:
-                pubmed_data[k] = temp
-                pubmed_metadata[k] = metadata
-            print(src)
-        except:
-            if verbose:
-                print('Error: %s' % src)
-            pass
-    dst = settings.metadata_dir / 'pubmed.pkl'
-    with open(dst, 'w') as f:
-        pickle.dump(pubmed_metadata, f)
-    return pubmed_data
 
+def parse_pdf(src):
+    """Parse research paper PDF and return text."""
+    try:
+        # try to open file
+        with open(src, 'rb') as f:
+            doc = slate.PDF(f)
 
-def filter_arxiv(output, arxiv_drops):
-    filtered_output = OrderedDict()
-    for k, v in output.items():
-        if v['metadata']['source'] == 'arxiv':
-            for term in v['metadata']['tags']:
-                if term['term'] not in arxiv_drops:
-                    filtered_output[copy.deepcopy(k)] = copy.deepcopy(v)
-        else:
-            filtered_output[copy.deepcopy(k)] = copy.deepcopy(v)
-    return filtered_output
+        # get text: strip out newlines and extra spaces
+        doc = ' '.join([' '.join(x.split()) for x in doc])
+        text = (doc
+            .split(' Abstract ')[-1]
+            .split(' Acknowledgments ')[0]
+            .split(' ∗ ∗ ∗ ')[0]
+            .strip()
+        )
 
+    except:  #  PDFSyntaxError
+        text = None
 
-def save_outputs(output, dst_dir=None, fn_prefix=None):
-    if dst_dir is None:
-        dst_dir = settings.processed_dir / 'data'
-    utilities.mkdir(dst_dir)
-    fn = 'data.json'
-    if fn_prefix is not None:
-        fn = fn_prefix + '_' + fn
-    dst = dst_dir / fn
-    with open(dst, 'w') as f:
-        json.dump(output, f)
+    return text
 
 
-def main():
+def collect_process_main():
```

Comment:
Not a big deal, but if these functions have been rearranged alphabetically should collect_process_main() go up under aggregate_pubmed()?

Reply:
It is a big deal! Fixed.